startup/92-optimization.py

```diff
@@ -1,70 +1,124 @@
-# !!! det reading: 343.0██████████████████████████████████                                                                 | 0.1384375/0.3955 [00:00<00:01,  6.09s/mm]
-# !!! det reading: 230.0█████████████████████████████████████████▉                                                           | 0.16625/0.3955 [00:00<00:01,  6.00s/mm]
-# !!! det reading: 196.0████████████████████████████████████████████████▉                                                    | 0.19375/0.3955 [00:01<00:01,  5.94s/mm]
-# !!! det reading: 89.0█████████████████████████████████████████████████████████                                             | 0.22125/0.3955 [00:01<00:01,  5.90s/mm]
-# !!! det reading: 79.0██████████████████████████████████████████████████████████████████▋                                   | 0.25875/0.3955 [00:01<00:00,  5.85s/mm]
-# !!! det reading: 201.0████████████████████████████████████████████████████████████████████████▊                            | 0.28625/0.3955 [00:01<00:00,  5.83s/mm]
-# !!! det reading: 60.0████████████████████████████████████████████████████████████████████████████████▉                     | 0.31375/0.3955 [00:01<00:00,  5.81s/mm]
-# !!! det reading: 124.0███████████████████████████████████████████████████████████████████████████████████████              | 0.34125/0.3955 [00:01<00:00,  5.79s/mm]
-# !!! det reading: 156.0██████████████████████████████████████████████████████████████████████████████████████████████       | 0.36875/0.3955 [00:02<00:00,  5.77s/mm]
-# !!! det reading: 205.0████████████████████████████████████████████████████████████████████████████████████████████████████▌| 0.39375/0.3955 [00:02<00:00,  5.80s/mm]
-# !!! det reading: 205.025mm [00:02,  6.33s/mm]
-# New stream: 'tes_hardware_flyer'
-
-# In [7]: hdr = db['7744d3c6-7640-42e7-ae43-a404069cc181']
-#
-# In [8]: hdr.table(stream_name='tes_hardware_flyer', fields=['tes_hardware_flyer_sample_stage_x_position', 'tes_hardware_flyer_sample_stage_y_position', 'tes_hardwar
-#    ...: e_flyer_intensity'])
-# Out[8]:
-#                                  time  tes_hardware_flyer_intensity  tes_hardware_flyer_sample_stage_x_position  tes_hardware_flyer_sample_stage_y_position
-# seq_num
-# 1       2020-08-05 14:33:14.213995934                         205.0                                   54.365938                                   40.369062
-#
-# In [9]:
-
-# ^^^ solved by rewriting the watch function (no globals)
-
-# TODO:
-# 1) have a live plot for the fitness (convergence plot)
-# 2) improve the algorithm to optimize the travel path (should not go back to the same point)
-# 3) Usability:
-#    - convenient I/O to the saved data (pandas dataframes, sorting, ...)
-#    - clean up prints and better output log
-#    - add metadata about the samples
-#    - plot trajectory
+import sys
+import textwrap
+import traceback
+
+from tabulate import tabulate
+
+
+def _extract_tb():
```

Comment:
A short explanation of what this function does would be helpful to the reader.

Reply:
I added a docstring via 0704c6e4f8d177bf7ac97667dc48e46b94d182da.